### ISSUE TITLE
Hide the quoted email when display a reply

### ISSUE THREAD

**Issue by DenBond7** [FlowCrypt/flowcrypt-android#2902]
I also noticed that for every reply ever received in the thread conversation view, the quoted email is permanently shown, whether it is an indented blockquote or uses ">" for the quoted email.

![image](https://github.com/user-attachments/assets/ff897bfd-369c-492a-9c84-6c1ec4dec89a)

Meanwhile, Gmail and the FlowCrypt browser extension hide the previous conversations/quoted emails and provide a triple-dot icon button that, when clicked, reveals the previous conversation. I just thought it might be something missing or an improvement for UX to help users focus on the important replies first.

![image](https://github.com/user-attachments/assets/84d2f96a-a0a7-45d9-bce8-fcb094e6c281)

![image](https://github.com/user-attachments/assets/dc8b2bcc-4fa2-40f2-8bc6-03bf66a4e1cd)

_Originally posted by @martgil in https://github.com/FlowCrypt/flowcrypt-android/issues/2821#issuecomment-2510515736_

**Comment by DenBond7** [comment]
@sosnovsky @ioanmo226 

I've found how you parse quotes

https://github.com/FlowCrypt/flowcrypt-ios/blob/72cb26b61786a736f91f6dad2c0a42d655410237/FlowCrypt/Functionality/Mail%20Provider/Message%20Provider/ProcessedMessage.swift#L132

Could you tell me how you hide/show quoted text? Do you use JavaScript?

**Comment by sosnovsky** [comment]
No, it's done natively - we toggle `shouldShowQuote` flag when user taps  `...` button.
And depending on `shouldShowQuote` value we show/hide `quoteNode`

https://github.com/FlowCrypt/flowcrypt-ios/blob/72cb26b61786a736f91f6dad2c0a42d655410237/FlowCryptUI/Cell%20Nodes/MessageTextSubjectNode.swift#L80

**Comment by DenBond7** [comment]
> And depending on shouldShowQuote value we show/hide quoteNode

```
private let messageNode = ASEditableTextNode()
private let quoteNode = ASEditableTextNode()
```

`messageNode` or `quoteNode` - is it a view(part of UI, Sorry, I don't know the right terminology) that can show HTML?

You have a separate view to show `base content` and `quoted text`. Right?

**Comment by ioanmo226** [comment]
Yeah, `messageNode` and `quoteNode` is something like `TextView` which shows message content and quote content.
When user clicks `toggleQuoteButtonNode`, then it toggles `quoteNode`

**Comment by DenBond7** [comment]
One more question

On Android, we sanitize received content. We have the following

https://github.com/FlowCrypt/flowcrypt-android/blob/834985515e9c13e987a363abfbbff4bd8fe5929d/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpMsg.kt#L128

https://github.com/FlowCrypt/flowcrypt-android/blob/834985515e9c13e987a363abfbbff4bd8fe5929d/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpMsg.kt#L166

https://github.com/FlowCrypt/flowcrypt-android/blob/834985515e9c13e987a363abfbbff4bd8fe5929d/FlowCrypt/src/main/java/com/flowcrypt/email/security/pgp/PgpMsg.kt#L178

Do you have something similar?

Why am I asking? I have to use `WebView` to show received content because we allow some tags that I can't show inside `TextView`. Just interesting what is permitted on iOS.

**Comment by ioanmo226** [comment]
We sanitize html in javascript side.

https://github.com/FlowCrypt/flowcrypt-ios/blob/72cb26b61786a736f91f6dad2c0a42d655410237/FlowCrypt/Functionality/Mail%20Provider/Message%20Provider/ProcessedMessage.swift#L118

https://github.com/FlowCrypt/flowcrypt-ios/blob/72cb26b61786a736f91f6dad2c0a42d655410237/Core/source/mobile-interface/endpoints.ts#L123

**Comment by DenBond7** [comment]
For me

https://stackoverflow.com/questions/10813581/can-i-replace-the-expand-icon-of-the-details-element
https://stackoverflow.com/questions/55031395/how-to-change-summary-text-if-details-has-open-attr
https://www.sitepoint.com/style-html-details-element/

**Comment by DenBond7** [comment]
@sosnovsky @ioanmo226 One more question. How do you identify quotes for encrypted messages?

**Comment by sosnovsky** [comment]
We have simple check for finding lines which start from `On ` and end with ` wrote:` - https://github.com/FlowCrypt/flowcrypt-ios/blob/72cb26b61786a736f91f6dad2c0a42d655410237/FlowCrypt/Functionality/Mail%20Provider/Message%20Provider/ProcessedMessage.swift#L160

**Comment by DenBond7** [comment]
#2933 covered only the `HTML` version

We need to cover the `plain text` version like

```eml
Message-ID: <ZE3RJQg@mail.flowcrypt.test>
Subject: Re: quotes
To: Den <den@flowcrypt.test>
Content-Type: text/plain; charset="UTF-8"
Content-Transfer-Encoding: quoted-printable

Reply from Android

Regards,

Den at FlowCrypt

On 2025-02-20 at 13:13, Den<denbond7@flowcrypt.test> wrote:
> rrr
>
> =D1=87=D1=82, 20 =D0=BB=D1=8E=D1=82. 2025=E2=80=AF=D1=80. =D0=BE 13:12 De=
n at FlowCrypt <den@flowcrypt.test> =D0=BF=D0=B8=D1=88=D0=B5:
> >
> > Reply 1
> >
> >
> > On Thu, Feb 20, 2025 at 1:10=E2=80=AFPM Den<denbond7@flowcrypt=
l.test> wrote:
> > >
> > > eee
> > >
> > > --
> > > Regards,
> > > Den
> >
> >
> >
> > --
> > Regards,
> > Den at FlowCrypt
>
>
>
> --
> Regards,
> Den
```

for a separate PR

**Comment by DenBond7** [comment]
All cases are covered. I'm closing this issue.
